picoscope/picobase.py

```diff
@@ -335,6 +335,21 @@ def getMaxMemorySegments(self):
         segments = self._lowLevelGetMaxSegments()
         return segments
 
+    def setExtTriggerRange(self, VRange=0.5):
+        """ This function sets the range for the EXT trigger channel
+            VRange MUST either be 0.5 or 5 for +/- 500mV or 5V ranges
+            respectively
+            This is only implemented for PS4000 series devices.
+        """
+        if VRange == 0.5:
+            VRangeApiVal = 5
+        elif VRange == 5:
+            VRangeApiVal = 8
+        else:
+            raise IOError("External trigger VRange is not 0.5 or 5")
```

Comment:
Also, don't bother checking for out of range errors too hard.
Maybe check for errors in the same way it is done in setChannel

Reply:
The error checking done in setChannel doesn't apply here. This is not like the setChannel range. Only api range values 5 and 8 are valid.

Reply:
I guess I just don't think it is too necessary that you check it for any scope. 

Their SDK does quite a bit of range checking from what I remember and whatever you add to this python module, brings you one step closer to getting an answer like:

> we don't support colinoflynn/pico-python

on their forums.

If you let their SDK do the error checking, you are closer to being able to generalize the code to other scopes.